### PR TITLE
support metric namespace

### PR DESCRIPTION
**Pull request by Nicholaswang**
## Description

support metric namespace option in config

## Pull request type
Please check the type of change your PR introduces:
- [ ] Feature

### Checklist

- [ ] Linter passes correctly
- [ ] Add tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

## Does this introduce a breaking change?

- [ ] No

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -148,55 +203,72 @@ var (
...
+		Name:      "bad_requests_total",
+		Help:      "Total number of unsupported requests",
 	})
-)
+}
+
+func getNamespace(cfg *config.Config) string {
+	namespace := ""
+	if cfg.EnableMetricNamespace {
+		namespace = "chproxy"
```
Comment (sigua-cs) [inline metrics.go]: IMO, this field can be configurable. This `chproxy` value can be the default. If someone needs to use a different value than the default, they will be able to set it in the config.

Comment (Nicholaswang) [reply metrics.go]: Actually, I agree with you. Would it be better to use another option MetricNamespace `string`?

Comment (Nicholaswang) [reply metrics.go]: Or providing EnableMetricNamespace `bool` and MetricNamespace `string` together?

Comment (sigua-cs) [reply metrics.go]: Yes. I think so

Comment (Nicholaswang) [reply metrics.go]: To confirm：
1）providing EnableMetricNamespace `bool` and MetricNamespace `string` combined
2)  providing MetricNamespace `string` only
Which option do you prefer? @sigua-cs

**Comment by Nicholaswang** [comment]
adding Namespace  `string` option in Server/Metrics struct

**Comment by sigua-cs** [comment]
Hello @Nicholaswang , I reviewed the PR and tested it locally. Looks good.
One more thing I would ask you to add. Please add to the `testdata` config example, which includes the `namespace` entity.

**Comment by Nicholaswang** [comment]
> Hello @Nicholaswang , I reviewed the PR and tested it locally. Looks good. One more thing I would ask you to add. Please add to the `testdata` config example, which includes the `namespace` entity.

sure

**Review by mga-chka — APPROVED** [review]

**Review by sigua-cs — APPROVED** [review]

**Pull request merged**